core/src/main/java/com/novoda/noplayer/internal/exoplayer/NoPlayerAdsLoader.java

```diff
@@ -0,0 +1,131 @@
+package com.novoda.noplayer.internal.exoplayer;
+
+import android.os.Handler;
+import android.os.Looper;
+import android.support.annotation.Nullable;
+
+import com.google.android.exoplayer2.Player;
+import com.google.android.exoplayer2.Timeline;
+import com.google.android.exoplayer2.source.ads.AdPlaybackState;
+import com.google.android.exoplayer2.source.ads.AdsLoader;
+import com.novoda.noplayer.AdvertBreak;
+import com.novoda.noplayer.AdvertsLoader;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class NoPlayerAdsLoader implements AdsLoader, Player.EventListener {
+
+    private final AdvertsLoader loader;
+    @Nullable
+    private Player player;
+    @Nullable
+    private AdPlaybackState adPlaybackState;
+    @Nullable
+    private EventListener eventListener;
+
+    private int adIndexInGroup = -1;
+    private int adGroupIndex = -1;
+
+    NoPlayerAdsLoader(AdvertsLoader loader) {
+        this.loader = loader;
+    }
+
+    @Override
+    public void setSupportedContentTypes(int... contentTypes) {
+        // no-op
+    }
+
+    @Override
+    public void start(final EventListener eventListener, AdViewProvider adViewProvider) {
+        this.eventListener = eventListener;
+        if (adPlaybackState == null) {
```

Comment:
I think we need to call the updatePlaybackState in case it is not null to resume ads setup with the new playback. It can happen and is our responsibility to handle. Good catch.

Reply:
https://github.com/novoda/no-player/pull/204/commits/272f9ec3bafeef732e8ff95f34f9705a6d270d8f